code/AssetLib/3DS/3DSLoader.cpp

```diff
@@ -156,9 +156,9 @@ void Discreet3DSImporter::InternReadFile(const std::string &pFile,
     mScene = &_scene;
 
     // Initialize members
-    D3DS::Node _rootNode("UNNAMED");
+    D3DS::Node *_rootNode = new D3DS::Node("UNNAMED");
     mLastNodeIndex = -1;
-    mCurrentNode = &_rootNode;
+    mCurrentNode = _rootNode;
     mRootNode = mCurrentNode;
```

Comment:
I don't _quite_ understand this code, but it does not make sense regardless.
Taking a pointer to a local variable is _always_ bad.
Somewhere down below, these pointers (except `mCurrentNode`) gets invalidated to `nullptr` but only in debug...for some reason.

Reply:
No, there are valid reasons to take an address of a local variable. In this case it appears that `_rootNode` is used as a temporary container and in the end its contents are copied into the final scene. Your changes on the other hand caused a memory leak because you didn't delete the object you created.

Reply:
After some rummaging, I find that `mRootNode` is indeed just a temporary variable. Although then its probably better to not make them a member variable?) On the other hand, `mCurrentNode` will get assigned a value here:

https://github.com/assimp/assimp/blob/1427e67b54906419e9f83cc8625e2207fbb0fcd5/code/AssetLib/3DS/3DSLoader.cpp#L636-L643

and slightly below that,

https://github.com/assimp/assimp/blob/1427e67b54906419e9f83cc8625e2207fbb0fcd5/code/AssetLib/3DS/3DSLoader.cpp#L648-L676

The second one seems correct enough, assuming it gets cleanup later as you said. But the first one is suspicious, because `FindNode` returns a pointer to the temporary variable `mRootNode` in this case.

https://github.com/assimp/assimp/blob/1427e67b54906419e9f83cc8625e2207fbb0fcd5/code/AssetLib/3DS/3DSLoader.cpp#L561-L574

As a result, I suggest adding an assert at the end of the function where `_rootNode` to verify that `mCurrentNode` have been assigned something else. This will not solve all the problems if it returns something in the child. I suppose one could introduce a more robust assertion that traverse through the entire graph.